addons/mail/static/src/core/common/store_service.js

```diff
@@ -446,7 +446,29 @@ export class Store extends BaseStore {
     }
 
     /** Provides an override point for when the store service has started. */
-    onStarted() {}
+    onStarted() {
+        navigator.serviceWorker.addEventListener("message", ({ data }) => {
+            const { type, payload } = data || {};
+            if (type !== "notification-display-request") {
+                return;
+            }
+            const { correlationId, model, res_id } = payload;
+            const thread = this.Thread?.get?.({ model, id: res_id });
+            let isTabFocused;
+            try {
+                isTabFocused = parent.document.hasFocus();
+            } catch {
+                // noop
```

Comment:
Maybe add some context (getting parent document from iframe result in CORS error).